### PR TITLE
GEOPY-1529: Freeze parameters and update InputFile data on BaseData construction

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-1529 - Freeze parameters and update InputFile data on BaseData construction**

### CODE REVIEW

File: curve_apps/driver.py [hunk 1]
```diff
@@ -87,17 +86,15 @@ def params(self, val: BaseData):
...
 
         with tempfile.TemporaryDirectory() as temp_dir:
-            file_path = self.params.input_file.write_ui_json(
-                path=temp_dir, name=self.params.name + ".ui.json"
-            )
-            entity.add_file(str(file_path))
+            filepath = Path(temp_dir) / f"{self.params.name}.ui.json"
+            self.params.write_ui_json(filepath)
+
+            entity.add_file(str(filepath))
```
Comment (domfournier) [inline curve_apps/driver.py:100]: Should try to streamline this on https://mirageoscience.atlassian.net/browse/GEOPY-1536

**Review by domfournier — APPROVED** [review]
Looks good.

File: curve_apps/driver.py (hunk 1 above)
Comment (benk-mira) [reply curve_apps/driver.py:100]: yes, sounds good

**Review by domfournier — APPROVED** [review]

**Review by domfournier — APPROVED** [review]

**Review by domfournier — APPROVED** [review]

File: curve_apps/utils.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+from __future__ import annotations
+
+import numpy as np
+from scipy.spatial import Delaunay
+
+from curve_apps.trend_lines.params import DetectionParameters
+
+
+def find_curves(
```
Comment (MatthieuCMira) [inline curve_apps/utils.py:16]: Thoses functions are quite complexe. 
Cutting them in sub function could help clarity

**Review by MatthieuCMira — APPROVED** [review]
minor comment

File: curve_apps/utils.py (hunk 2 above)
Comment (domfournier) [reply curve_apps/utils.py:16]: let's do it here. Need those changes to go forward for other issues
https://mirageoscience.atlassian.net/browse/GEOPY-1537

**Pull request merged**